src/components/memorygame/MemoryGameCards/index.jsx

```diff
@@ -0,0 +1,163 @@
+  const [t] = useTranslation();
+  // cardsDataArr example [{description:"...",link:"...",phrase:".."},...]
+  const cardsDataArr = [
+    ...t("memoryGame.cards", {
+      returnObjects: true,
+    }),
+  ];
+  // cardStates is an array of states with multiple properties
+  // state example :
+  // {cardKey:"",img:{src:"",imgKey:""},
+  // isFlipped:false,isMatched:false,description:"",link:"",phrase:""}
+  const [cardStates, setCardStates] = useState(
+    cardInitialStateGenerator(cardsDataArr, NUM_REQUIRED_MATCHES)
+  );
+  // selectedCards is an array of clicked card states(objects).
+  // selectedCards array hold only "NUM_REQUIRED_MATCHES" amount of card states
+  const [selectedCards, setSelectedCards] = useState([]);
+
+  // handle click flips card
+  // if the card is not in selectedCards, adds it to selectedCards
+  const handleClick = (clickedCardState) => {
+    const copyOfCardStates = [...cardStates];
+    // current index is needed to keep the order of cards displayed
+    // because goal is to make user remember the spots of previous cards
+    const currentIndex = cardStates.findIndex(
+      (state) => state.cardKey === clickedCardState.cardKey
+    );
+    const targetCardState = copyOfCardStates[currentIndex];
+    targetCardState.isFlipped = true;
+    setCardStates(() => [...copyOfCardStates]);
+    // prevents user to use same card multiple times
+    const isCardNotInSelectedCards = selectedCards.every(
+      (selectedCard) => selectedCard.cardKey !== clickedCardState.cardKey
+    );
+    if (isCardNotInSelectedCards) {
+      setSelectedCards((prevState) => [...prevState, clickedCardState]);
+    }
+  };
+
+  // Checks if two cards have a matching image.
+  const areCardsMatching = (card, firstCard) => {
+    return card.img.imgKey === firstCard.img.imgKey;
+  };
+
+  // manipulates isFlipped and isMatched property of card state
+  const changeCardStatePropertyToOpposite = (property, targetCard) => {
+    const copyOfCardStates = [...cardStates];
+    const targetIndex = copyOfCardStates.findIndex(
+      (cardState) => cardState.cardKey === targetCard.cardKey
+    );
+    const targetCardState = copyOfCardStates[targetIndex];
+    // isMatched is false as default. When it is true success symbol appears in card
+    if (property === "isMatched") {
+      targetCardState.isMatched = !targetCardState.isMatched;
+      setCardStates(() => [...copyOfCardStates]);
+      // matchedCards hold successfuly matched card states and uses them to display facts
+      // to prevent fact duplication only 1 of the matched cards should be passed into matchedCards
+      setMatchedCards((prevState) => [...prevState, selectedCards[0]]);
+      // isFlipped is false as default. When it is true backside of card become visible
+    } else if (property === "isFlipped") {
+      targetCardState.isFlipped = !targetCardState.isFlipped;
+      setCardStates(() => [...copyOfCardStates]);
+    }
```

Comment:
else throw an error.. defensively program here. otherwise if someone passes the wrong thing or makes a typo in property your code will simply run

Reply:
` else {
            throw new Error('invalid property');
        }`

Reply:
PTAL